core/commands/cat.go

```diff
@@ -96,8 +98,15 @@ var CatCmd = &cmds.Command{
 
 				switch val := v.(type) {
 				case io.Reader:
-					bar, reader := progressBarForReader(os.Stderr, val, int64(res.Length()))
-					bar.Start()
+					reader := val
+
+					req := res.Request()
+					progress, _ := req.Options[progressOptionName].(bool)
+					if progress {
+						var bar *pb.ProgressBar
+						bar, reader = progressBarForReader(os.Stderr, val, int64(res.Length()))
+						bar.Start()
```

Comment:
does this need a `bar.Finish()` somewhere?

Reply:
Wouldn't hurt a defer.